### PR TITLE
Feature/rest api

### PR DESCRIPTION
**Pull request by Brewer-Slack**
Basic framework for a rest api using Django's rest framework. I'll get started learning more about it this week.
Navigation to the user page is localhost:8000/users/

### CODE REVIEW

File: website/yw_website/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "/usr/local/bin/python3"
+}
```
Comment (clundin55) [inline website/yw_website/.vscode/settings.json:3]: Make sure you don't push your vscode files. Tony and I are trying to fix the git ignore but you'' likely get a lot of comments relating to this.

File: website/yw_website/rest/apps.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+
+class QuickstartConfig(AppConfig):
```
Comment (clundin55) [inline website/yw_website/rest/apps.py:4]: Might be better to call it RestConfig, Quickstart is a vague name

File: website/yw_website/yw_website/urls.py [hunk 3]
```diff
@@ -19,10 +19,33 @@
...
+router = routers.DefaultRouter()
+router.register(r'users', UserViewSet)
+
+
 urlpatterns = [
     path('admin/', admin.site.urls),
     path('', views.home, name='home'),
-    path('upload/$', views.model_form_upload, name='upload')
+    path('upload/', views.model_form_upload, name='upload'),
+    url(r'^', include(router.urls)),
```
Comment (clundin55) [inline website/yw_website/yw_website/urls.py:48]: Change url to use path instead, as url is set to be deprecated soon.
https://stackoverflow.com/questions/47947673/is-it-better-to-use-path-or-url-in-urls-py-for-django-2-0

**Review by clundin55 — CHANGES_REQUESTED** [review]
Looks good! A couple minor changes but otherwise well done! Make sure you pull --rebase as this PR is behind master.

File: website/yw_website/.vscode/settings.json (hunk 1 above)
Comment (aniehuser) [inline website/yw_website/.vscode/settings.json:3]: End files with a [newline](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)

File: website/yw_website/rest/serializers.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+class UserSerializer(serializers.HyperlinkedModelSerializer):
+    class Meta:
+        model = User
+        fields = ('url', 'username', 'email', 'groups')
+
+
+class GroupSerializer(serializers.HyperlinkedModelSerializer):
+    class Meta:
+        model = Group
+        fields = ('url', 'name')
```
Comment (aniehuser) [inline website/yw_website/rest/serializers.py:13]: newline

File: website/yw_website/.vscode/settings.json (hunk 1 above)
Comment (clundin55) [reply website/yw_website/.vscode/settings.json:3]: This shouldn't be committed anyway. But a good point nonetheless

**Comment by clundin55** [comment]
Fixed merge conflicts with master. You can merge after implementing the requested changes.

File: yw_website/yw_website/urls.py [hunk 5]
```diff
@@ -19,10 +19,37 @@
...
+
+# Routers provide an easy way of automatically determining the URL conf.
+router = routers.DefaultRouter()
+router.register(r'users', UserViewSet)
+
+
 urlpatterns = [
     path('admin/', admin.site.urls),
     path('', views.home, name='home'),
+<<<<<<< HEAD:website/yw_website/yw_website/urls.py
```
Comment (aniehuser) [inline yw_website/yw_website/urls.py:46]: fix this merge conflict

**Pull request closed without merging**